### PR TITLE
Remove Devise

### PR DESCRIPTION
**Pull request by monfresh**
**Why**:
- Devise is not needed in order to support Omniauth authentication.

**How**:
- Remove the Devise gem and related files
- Remove Devise-related DB columns
- Move the Omniauth configuration from the Devise config to a new
initializer
- Use `SessionsController` to handle the authentication
- Add a `success` page to send the user there after successful
authentication, instead of prompting the user to sign in again on the
home page
- Use the correct `saml_test_sp.key` (it was erroneously modified
during a rebase with master)

### CODE REVIEW

**Comment by monfresh** [comment]
cc @ertzeid if you want to see how I removed Devise.

File: spec/controllers/sessions_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  describe 'POST /auth/saml/callback' do
+    context 'when the authorization is valid' do
+      it 'signs the user in and redirects to dashboard with flash notice' do
+        configure_valid_omniauth_login
+        post :create, provider: :saml
+
+        expect(session[:user_id]).to eq User.last.id
+        expect(response).to redirect_to success_url
+        expect(flash[:notice]).
+          to eq I18n.t('omniauth_callbacks.success')
```
Comment (sbc100) [inline spec/controllers/sessions_controller_spec.rb]: Looks like it will fit on one line?

**Comment by monfresh** [comment]
@sbc100 I addressed your comment (putting the expectation all on one line). Was that the only thing?

**Comment by sbc100** [comment]
lgtm, but maybe @ertzeid should take a look too?

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  def failure
+    flash[:error] = t('omniauth_callbacks.failure', reason: failure_message)
+    redirect_to root_url
+  end
+
+  private
+
+  def failure_message
+    env['omniauth.error.type'].to_s.humanize
```
Comment (ertzeid) [inline app/controllers/sessions_controller.rb]: Will this provide any sort of sensitive information to end users? This came up in the original PR and I switched to a generic "Login failed, please try again later"-type message on failure.

**Comment by ertzeid** [comment]
Just the one question, but overall looks good.

File: app/controllers/sessions_controller.rb (hunk 2 above)
Comment (monfresh) [inline app/controllers/sessions_controller.rb]: Not that I know of. The only examples of error types I can find are `Invalid Credentials` and `Invalid Ticket`. I'm not sure how we could simulate a plausible error to see what kind of message is displayed.

Comment (monfresh) [inline app/controllers/sessions_controller.rb]: I'm fine with either the generic message or the one based on the omniauth error, which seems to be the convention based on what Devise does and other documentation I've seen.

**Comment by sashax** [comment]
LGTMT, now that that last q has been addressed.

**Pull request merged**